app/components/downloaders/access_summary_table_data_component.rb

```diff
@@ -34,31 +37,31 @@ def icon
     private
 
     def grantee_name
-      @other_org.name
+      @grantee_org.name
     end
 
     def grantor_name
-      @organization.name
+      @grantor_org.name
     end
 
     def group_memberships_text
-      (@organization.downloader_groups & @other_org.groups).map(&:display_name).to_sentence
+      (@grantor_org.downloader_groups & @grantee_org.groups).map(&:display_name).to_sentence
     end
 
     def can_download?
-      unrestricted_access? || @other_org.effective_downloadable_organizations.include?(@organization)
+      unrestricted_access? || @grantee_org.effective_downloadable_organizations.include?(@grantor_org)
```

Comment:
Since we always show "ORG is not a provider" for non-providers, should we also check for that here in the weird case where somehow a non-provider has restrict_download flipped to false? Probably wouldn't/shouldn't happen...

Reply:
Yeah, that's a weird case, I think. Should we suppress the whole column (and access controls) for non-providers (64f6bca)?